src/uri.jl

```diff
@@ -0,0 +1,20 @@
+function URIParser.URI(p::AbstractPath)
+    if isempty(root(p))
+        error("$p is not an absolute path")
```

Comment:
Probably want to throw an `ArgumentError` here and we should test that error condition.